frontend/src/components/member_input.tsx

```diff
@@ -0,0 +1,155 @@
+// Author: Andrew Lockard
+import React, { useState, useEffect } from "react";
+
+interface TeamMember {
+  id: number | null;
+  first_name: string;
+  last_name: string;
+}
+
+export const MemberInput = ({ token }: { token: string }) => {
+  const [members, setMembers] = useState<TeamMember[]>([]);
+  const [newName, setNewName] = useState<string>("");
+  const [errorDisplay, setErrorDisplay] = useState<JSX.Element>(<></>);
+
+  useEffect(() => {
+    const getMembers = async () => {
+      const res = await fetch("/api/team/members", {
+        headers: { Authorization: `Bearer ${token}` },
+      });
+      const json = (await res.json()) as TeamMember[];
+      setMembers(json);
+    };
+    void getMembers();
+  }, [token]);
+
+  const delete_member = async (member_id: number) => {
+    const res = await fetch(`/api/team/members/${member_id}`, {
+      method: "DELETE",
+      headers: { Authorization: `Bearer ${token}` },
+    });
+    if (res.ok) {
+      setMembers((members) =>
+        members.filter((member) => member.id !== member_id)
+      );
+    }
+  };
+
+  const add_new_member = async () => {
+    const [firstname, ...rest] = newName.split(" ");
+    const lastname = rest.join(" ");
+    if (firstname === "" || lastname === "") {
+      throw new Error("Please enter a name.");
```

Comment:
Nit: Error handling could be more descriptive... If a student types "Nick" and is told to enter a name, i'm sure they would be confused. 

Update: Looking at the implementation, I see greater detail from another error handler

Reply:
Actually I agree with you here. As just typing "Nick" would result in it saying to "Please type a name" since that if statement would catch it.